lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -287,7 +287,7 @@ static bool GetUnsignedVal(Value *V, uint32_t *pValue) {
   return true;
 }
 
-static void MarkUsedSignatureElements(Function *F, DxilModule &DM) {
+static void MarkUsedSignatureElements(Function *F, DxilModule &DM, bool IsLib) {
```

Comment:
I know what your problem was: it was the patch constant shader case.  That could be solved more elegantly by passing the entry props to this function.  For the hull shader case outside this function, we would simply pass the hull shader props along with the patch constant function.  That would allow you to avoid all the special casing below.